### PR TITLE
feat(debug): add checkpoint debug events

### PR DESCRIPTION
**Pull request by dqbd**
Fixes # (issue)

### CODE REVIEW

File: langgraph/src/pregel/algo.ts [hunk 1]
```diff
@@ -417,17 +418,14 @@ export function _prepareNextTasks<
         });
       }
     } else {
-      taskDescriptions.push({
-        name: packet.node,
-        input: packet.args,
-      });
+      taskDescriptions.push({ id: taskId, name: packet.node });
```
Comment (jacoblee93) [inline langgraph/src/pregel/algo.ts:421]: No need for `input` here? Was a mistake before?

File: langgraph/src/pregel/algo.ts
Comment (jacoblee93) [inline langgraph/src/pregel/algo.ts:1]: This all looks like just pure refactoring right?

File: langgraph/src/pregel/debug.ts [hunk 2]
```diff
@@ -128,13 +142,51 @@ export function* mapDebugTaskResults<
...
   channels: Record<string, BaseChannel>,
   streamChannels: string | string[],
-  metadata: CheckpointMetadata
+  metadata: CheckpointMetadata,
+  tasks: readonly PregelExecutableTask<N, C>[],
+  pendingWrites: CheckpointPendingWrite[]
 ) {
+  function formatConfig(config: RunnableConfig) {
+    // https://stackoverflow.com/a/78298178
+    type CamelToSnake<
```
Comment (jacoblee93) [inline langgraph/src/pregel/debug.ts:159]: Nice

**Review by jacoblee93 — APPROVED** [review]

File: langgraph/src/pregel/algo.ts (hunk 1 above)
Comment (dqbd) [reply langgraph/src/pregel/algo.ts:421]: Haven't seen usage of `input` in both JS nor Python, thus removed it in favour of matching Python

File: langgraph/src/pregel/algo.ts
Comment (dqbd) [reply langgraph/src/pregel/algo.ts:1]: The goal was to add the `tasks` prop in debug checkpoint event, which required us to update the `PregelTaskDescription` with the `id`

**Pull request merged**